File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int res = 0;
        int toc = 0;
        if (ruleKey.equals("color")) toc = 1;
        else if (ruleKey.equals("name")) toc = 2;
        for (List<String> pair: items) {
            if (pair.get(toc).equals(ruleValue)) {
                res += 1;
            }
        }
        return res;
    }
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int res = 0;
        int toc = 0;
        if (ruleKey == "color") toc = 1;
        else if (ruleKey =="name") toc = 2;
        for (int i = 0; i < items.size(); i++) {
            if (items[i][toc] == ruleValue) res++;
        }
        return res;
    }
};